src/Symfony/Component/Messenger/Retry/MultiplierRetryStrategy.php

```diff
@@ -66,7 +66,7 @@ public function __construct(int $maxRetries = 3, int $delayMilliseconds = 1000,
 
     public function isRetryable(Envelope $message): bool
     {
-        if (0 === $this->maxRetries) {
+        if (null === $this->maxRetries) {
```

Comment:
Unrelated: changes behavior slightly so that `0` can be used to mean "don't retry on this transport". Previously, there was no easy way to say "don't retry" as 0 meant "retry infinitely".

Test added for this.